Crash/Events/CrashInit.cs

```diff
@@ -18,14 +19,44 @@ internal static class CrashInit
 
         internal static void OnInit(IEnumerable<Speck> specks)
         {
-            IsInit = true;
             Rhino.RhinoApp.WriteLine("Loading specks ...");
-            LocalCache.Instance.BakeSpecks(specks);
+
+            IsInit = true;
+            _HandleSpecks(specks);
             IsInit = false;
 
             Rhino.RhinoDoc.ActiveDoc.Views.Redraw();
         }
 
+        private static void _HandleSpecks(IEnumerable<Speck> specks)
+        {
+            var enumer = specks.GetEnumerator();
+            while(enumer.MoveNext())
+            {
+                _HandleSpeck(enumer.Current);
+            }
+        }
+
+        private static void _HandleSpeck(Speck speck)
+        {
+            LocalSpeck localSpeck = LocalSpeck.ReCreate(speck);
+            if (!speck.Temporary)
+            {
+                LocalCache.Instance.BakeSpeck(localSpeck);
+            }
+            else
+            {
+                if (speck.LockedBy == User.CurrentUser.name)
```

Comment:
I don't know if this would ever be called?  All locked specks _should_ be non-temporary..